### PR TITLE
[Dynamo] Fix graph break when `tensor.split()` is called within a device context manager

### LINKED ISSUE #139183: compile fails when split is called within a device cm
### 🐛 Describe the bug

The following code breaks when the compiled function is called within the context manager
```python
import torch

@torch.compile(fullgraph=True)
def split(x):
    return x.split(4, 0)

x = torch.randn(12)
split(x)
print('cpu passed')
x = torch.randn(12, device="cuda")
split(x)
print('cuda passed')

with torch.device("cuda"):
    x = torch.randn(12)
    split(x) # breaks
    print('cuda cm passed')
```

Dynamo seems to miss the `__torch_function__` here for some reason.

Error:
```
torch._dynamo.exc.Unsupported: Graph break due to unsupported builtin torch._VariableFunctionsClass.split. This function is either a Python builtin (e.g. _warnings.warn) or a third-party C/C++ Python extension (perhaps created with pybind). If it is a Python builtin, please file an issue on GitHub so the PyTorch team can add support for it and see the next case for a workaround. If it is a third-party C/C++ Python extension, please either wrap it into a PyTorch-understood custom operator (see https://pytorch.org/tutorials/advanced/custom_ops_landing_page.html for more details) or, if it is traceable, use torch.compiler.allow_in_graph.

from user code:
   File "/home/vmoens/script.py", line 5, in split
    return x.split(4, 0)
  File "/home/vmoens/.conda/envs/torchrl/lib/python3.11/site-packages/torch/utils/_device.py", line 106, in __torch_function__
    return func(*args, **kwargs)
  File "/home/vmoens/.conda/envs/torchrl/lib/python3.11/site-packages/torch/_tensor.py", line 1024, in split
    return torch._VF.split(self, split_size, dim)  # type: ignore[attr-defined]

Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information
```

### Versions

nightly

cc @hameerabbasi @rgommers @ezyang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### PR DESCRIPTION
**Pull request by shink**
Fixes: #139183

Note: this case can also be reproduced on cpu

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: test/dynamo/test_repros.py [hunk 1]
```diff
@@ -6195,6 +6195,18 @@ def fn(x):
         # Ensure that the listcomp is fully compiled
         self.assertEqual(cnt.op_count, 8)
 
+    # https://github.com/pytorch/pytorch/issues/139183
+    def test_tensor_split_within_device_cm(self):
+        @torch.compile(fullgraph=True)
+        def split(x):
+            return x.split(4, 0)
+
+        x = torch.randn(12)
```
Comment (vmoens) [inline test/dynamo/test_repros.py:6204]:
```suggestion
        x = torch.zeros(12)
```

File: test/dynamo/test_repros.py [hunk 2]
```diff
@@ -6195,6 +6195,18 @@ def fn(x):
         # Ensure that the listcomp is fully compiled
         self.assertEqual(cnt.op_count, 8)
 
+    # https://github.com/pytorch/pytorch/issues/139183
```
Comment (vmoens) [inline test/dynamo/test_repros.py:6198]:
```suggestion
```

**Review by vmoens — COMMENTED** [review]
Thanks for the quick fix!

**Comment by shink** [comment]
@ezyang Could you please help review? Thanks!

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**